app/src/main/java/com/github/se/studybuddies/MainActivity.kt

```diff
@@ -279,18 +275,58 @@ class MainActivity : ComponentActivity() {
-                  VideoCallScreen(videoCallViewModel, navigationActions)
-                  Log.d("MyPrint", "Successfully navigated to VideoGroupScreen")
+            composable(
+                route = "${Route.CALLLOBBY}/{groupUID}",
+                arguments = listOf(navArgument("groupUID") { type = NavType.StringType })) {
+                    backStackEntry ->
+                  val groupUID = backStackEntry.arguments?.getString("groupUID")
+                  if (groupUID != null && StreamVideo.isInstalled) {
+                    val viewModel = remember { CallLobbyViewModel(groupUID, callType) }
+                    Log.d("MyPrint", "Join VideoCall lobby")
+                    CallLobbyScreen(groupUID, viewModel, navigationActions)
+                  } else {
+                    Log.d("MyPrint", "Failed bc video call client isn't installed")
+                    navigationActions.navigateTo("${Route.GROUP}/$groupUID")
+                  }
                 }
-              } else {
-                navigationActions.goBack()
-              }
-            }
+
+            composable(
+                route = "${Route.VIDEOCALL}/{groupUID}",
+                arguments = listOf(navArgument("groupUID") { type = NavType.StringType })) {
+                    backStackEntry ->
+                  val groupUID = backStackEntry.arguments?.getString("groupUID")
+                  val streamVideo = StreamVideo.instance()
+                  val activeCall = streamVideo.state.activeCall.value
+                  if (groupUID != null) {
+                    val call =
+                        if (activeCall != null) {
+                          if (activeCall.id != groupUID) {
+                            Log.w("CallActivity", "A call with id: ${groupUID} existed. Leaving.")
+                            // If the call id is different leave the previous call
+                            activeCall.leave()
+                            // Return a new call
+                            streamVideo.call(callType, groupUID)
+                          } else {
+                            // Call ID is the same, use the active call
+                            activeCall
+                          }
+                        } else {
+                          // There is no active call, create new call
+                          streamVideo.call(callType, groupUID)
+                        }
+                    val videoVM = remember { VideoCallViewModel(call, groupUID) }
+                    Log.d("MyPrint", "Join VideoCallScreen")
+                    VideoCallScreen(
+                        call,
+                        videoVM,
+                    ) {
+                      videoVM.call.leave()
+                      StreamVideo.instance().state.activeCall.value?.leave()
+                      navController.popBackStack("${Route.GROUP}/$groupUID", false)
+                    }
+                  }
```

Comment:
It is a good idea to avoid "cascades" of tabs to improve readability, you can solve this problem using [early returrns](https://dev.to/jpswade/return-early-12o5) and function extraction (e.g., `startCall`)

```kotlin
 private fun startCall( /* ... */ ): Call {
   // ...
 }

  if (groupUID == null) { return }

  val call = startCall( /* ... */ )
  val videoVM = remember { VideoCallViewModel(call, groupUID) }
  Log.d("MyPrint", "Join VideoCallScreen")
  VideoCallScreen(
      call,
      videoVM,
  ) {
    videoVM.call.leave()
    StreamVideo.instance().state.activeCall.value?.leave()
    navController.popBackStack("${Route.GROUP}/$groupUID", false)
  }
```